styleguide/components/Type.tsx

```diff
@@ -10,16 +10,15 @@ export const styles = ({ fontFamily, fontSize, color }: Theme) => ({
   },
 });
 
-const KUBEVIRT_TYPES = KUBEVIRT_CONSTS.allTypes;
+const KUBEVIRT_TYPES = KUBEVIRT_CONSTS.kubevirtTypes;
 interface TypeProps extends JssInjectedProps {
   children: React.ReactNode;
 }
 
 function getGithubTypeHref(type: string): string {
-  return `https://github.com/kubevirt-ui/kubevirt-api/blob/main/kubevirt/models/${type.replace(
-    /\[\]/,
-    '',
-  )}.ts`;
+  return `https://github.com/kubevirt-ui/kubevirt-api/blob/main/${
+    KUBEVIRT_CONSTS.KUBEVIRT_MODEL_PATH
+  }/${type.replace(/\[\]/, '')}.ts`;
 }
 
 function TypeRenderer({ classes, children }: TypeProps) {
```

Comment:
```suggestion
const TypeRenderer: React.FC<TypeProps> = ({ classes, children }) => {
  const isWrappedWithLink = typeof children === 'string' && KUBEVIRT_TYPES.includes(children.replace(/\[\]/, ''))

  return <span className={classes.type}>{
  isWrappedWithLink ? 
  <a href={getGithubTypeHref(children)} target="_blank" rel="noreferrer">
        {children}
      </a> : 
      children}</span>;
}
```

Reply:
about functional components, there is a rule of Deepsource that does not like them: https://deepsource.io/gh/kubevirt-ui/kubevirt-components/run/465793fe-269b-427f-a88d-a2839956f249/javascript/JS-0399/
@glekner .
For this reason, they are all declared in the old fashion way :D

Reply:
disabled!